homeassistant/components/climate/opentherm_gw.py

```diff
@@ -0,0 +1,201 @@
+"""OpenTherm Gateway Climate component for Home Assistant."""
+
+import logging
+import voluptuous as vol
+
+from homeassistant.components.climate import (ClimateDevice, PLATFORM_SCHEMA,
+                                              STATE_IDLE, STATE_HEAT,
+                                              STATE_COOL,
+                                              SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.components.climate.modbus import CONF_PRECISION
+from homeassistant.const import (CONF_DEVICE, CONF_NAME, PRECISION_HALVES,
+                                 PRECISION_TENTHS, TEMP_CELSIUS,
+                                 PRECISION_WHOLE, STATE_ON, STATE_OFF,
+                                 STATE_UNKNOWN)
+from homeassistant.util.temperature import convert as convert_temperature
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pyotgw==0.1a0']
+
+CONF_FLOOR_TEMP = "floor_temperature"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DEVICE): cv.string,
+    vol.Optional(CONF_NAME, default="OpenTherm Gateway"): cv.string,
+    vol.Optional(CONF_PRECISION): vol.In([PRECISION_TENTHS, PRECISION_HALVES,
+                                          PRECISION_WHOLE]),
+    vol.Optional(CONF_FLOOR_TEMP, default=False): cv.boolean,
+})
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE)
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the opentherm_gw component."""
+    gateway = OpenThermGateway()
+    gateway.friendly_name = config.get(CONF_NAME)
+    gateway.floor_temp = config.get(CONF_FLOOR_TEMP, False)
+    gateway.temp_precision = config.get(CONF_PRECISION)
+    async_add_entities([gateway])
+    await gateway.connect(hass, config.get(CONF_DEVICE))
```

Comment:
Remove this. See comment below.

Reply:
Addressed in f6fb95fe6763bc9dac4b333e0c11a4c4ff85cfbb